l1-contracts/contracts/bridge/L1WethBridge.sol

```diff
@@ -151,42 +354,151 @@ contract L1WethBridge is IL1Bridge, ReentrancyGuard {
+
+            // Deposit WETH tokens from the depositor address to the smart contract address
+            IERC20(l1WethAddress).safeTransferFrom(msg.sender, address(this), _amount);
+            // Unwrap WETH tokens (smart contract address receives the equivalent amount of ETH)
+            IWETH9(l1WethAddress).withdraw(_amount);
         }
-        txHash = zkSync.requestL2Transaction{value: _amount + msg.value}(
-            l2Bridge,
-            _amount,
-            l2TxCalldata,
-            _l2TxGasLimit,
-            _l2TxGasPerPubdataByte,
-            new bytes[](0),
-            refundRecipient
-        );
+        {
+            if (!hyperbridgingEnabled[_chainId]) {
+                chainBalance[_chainId] += _mintValue;
+            }
+        }
+        {
+            // Request the finalization of the deposit on the L2 side
+            bytes memory l2TxCalldata = _getDepositL2Calldata(msg.sender, _l2Receiver, l1WethAddress, _amount);
+
+            // If the refund recipient is not specified, the refund will be sent to the sender of the transaction.
+            // Otherwise, the refund will be sent to the specified address.
+            // If the recipient is a contract on L1, the address alias will be applied.
+            address refundRecipient = _refundRecipient;
+            if (_refundRecipient == address(0)) {
+                refundRecipient = msg.sender != tx.origin
+                    ? AddressAliasHelper.applyL1ToL2Alias(msg.sender)
+                    : msg.sender;
+            }
+            txHash = _depositSendTx(
+                _chainId,
+                _mintValue,
+                _amount,
+                l2TxCalldata,
+                _l2TxGasLimit,
+                _l2TxGasPerPubdataByte,
+                refundRecipient
+            );
+        }
+        emit DepositInitiatedSharedBridge(_chainId, txHash, msg.sender, _l2Receiver, _l1Token, _amount);
+    }
 
-        emit DepositInitiated(txHash, msg.sender, _l2Receiver, _l1Token, _amount);
+    function _depositSendTx(
+        uint256 _chainId,
+        uint256 _mintValue,
+        uint256 _amount,
+        bytes memory _l2TxCalldata,
+        uint256 _l2TxGasLimit,
+        uint256 _l2TxGasPerPubdataByte,
+        address _refundRecipient
+    ) internal returns (bytes32 txHash) {
+        // we don't save the depositAmount because base asset is sent to refundrecipient
+
+        L2TransactionRequestDirect memory request = L2TransactionRequestDirect({
+            chainId: _chainId,
+            l2Contract: l2BridgeAddress[_chainId],
+            mintValue: _mintValue,
+            l2Value: _amount,
+            l2Calldata: _l2TxCalldata,
+            l2GasLimit: _l2TxGasLimit,
+            l2GasPerPubdataByteLimit: _l2TxGasPerPubdataByte,
+            factoryDeps: new bytes[](0),
+            refundRecipient: _refundRecipient
+        });
+        txHash = bridgehub.requestL2TransactionBaseTokenBridge(request);
```

Comment:
Since `L1WethBridge` is only for eth based chains, can you call legacy `requestL2Transaction` function instead? this way we can remove `requestL2TransactionBaseTokenBridge` completely

Reply:
Btw L1WethBridge is also for custom base token chains, just depositTx is only for eth based Chains. 

I can call the legacyL2Transaction method, but in that case the funds will be transferred back to this bridge. So it will be more expensve gas-wise. Wdyt?

Reply:
Ok, will just remove it